controllers/renovate_util.go

```diff
@@ -311,3 +313,146 @@ func CreateRenovaterJob(ctx context.Context, client client.Client, scheme *runti
+		renovateCmds = append(renovateCmds,
+			fmt.Sprintf("RENOVATE_TOKEN=$TOKEN_%d RENOVATE_CONFIG_FILE=/configs/%d.js renovate", installation.id, installation.id),
+		)
+	}
+	if len(renovateCmds) == 0 {
+		return nil
+	}
+	secret := &v1.Secret{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		StringData: secretTokens,
+	}
+	configMap := &v1.ConfigMap{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: configmaps,
+	}
+	trueBool := true
+	falseBool := false
+	renovateImageUrl := os.Getenv(RenovateImageEnvName)
+	if renovateImageUrl == "" {
+		renovateImageUrl = DefaultRenovateImageUrl
+	}
+	pipelineRun := &tektonapi.PipelineRun{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: tektonapi.PipelineRunSpec{
+			PipelineSpec: &tektonapi.PipelineSpec{
+				Tasks: []tektonapi.PipelineTask{{
+					Name: "renovate",
+					TaskSpec: &tektonapi.EmbeddedTask{
+						TaskSpec: tektonapi.TaskSpec{
+							Steps: []tektonapi.Step{{
+								Name:  "renovate",
+								Image: renovateImageUrl,
+								EnvFrom: []v1.EnvFromSource{
+									{
+										Prefix: "TOKEN_",
+										SecretRef: &v1.SecretEnvSource{
+											LocalObjectReference: v1.LocalObjectReference{
+												Name: name,
+											},
+										},
+									},
+								},
+								Command: []string{"bash", "-c", strings.Join(renovateCmds, "; ")},
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name:      name,
+										MountPath: "/configs",
+									},
+								},
+								SecurityContext: &v1.SecurityContext{
+									Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+									RunAsNonRoot:             &trueBool,
+									AllowPrivilegeEscalation: &falseBool,
+									SeccompProfile: &v1.SeccompProfile{
+										Type: v1.SeccompProfileTypeRuntimeDefault,
+									},
+								},
+							}},
+							Volumes: []v1.Volume{
+								{
+									Name: name,
+									VolumeSource: v1.VolumeSource{
+										ConfigMap: &v1.ConfigMapVolumeSource{
+											LocalObjectReference: v1.LocalObjectReference{Name: name},
+										},
+									},
+								},
+							},
+						},
+					},
+				}},
+			},
+		},
+	}
+	if debug {
+		pipelineRun.Spec.PipelineSpec.Tasks[0].TaskSpec.Steps[0].Env = append(pipelineRun.Spec.PipelineSpec.Tasks[0].TaskSpec.Steps[0].Env, v1.EnvVar{Name: "LOG_LEVEL", Value: "debug"})
+	}
+
+	if err := client.Create(ctx, secret); err != nil {
+		return err
+	}
+	if err := client.Create(ctx, configMap); err != nil {
+		return err
+	}
+	if err := client.Create(ctx, pipelineRun); err != nil {
+		return err
+	}
+	log.Info(fmt.Sprintf("Pipeline %s triggered", pipelineRun.Name), logs.Action, logs.ActionAdd)
+	if err := controllerutil.SetOwnerReference(pipelineRun, secret, scheme); err != nil {
+		return err
```

Comment:
I guess it's not possible to set the owner reference before `Create()`-ing the objects? Which means the secrets and configmaps may persist in the user namespace indefinitely (if e.g. the pipelineRun creation fails)?

Reply:
Actually it should be possible, because the PipelineRun will wait for creation, this is how the multi platform controller works. 

This code was copy pasted from the existing renovate code